doc/docs.md

```diff
@@ -1643,16 +1645,46 @@ fn main() {
 }
 ```
 
-### Matching sum types
+`as` will panic if `w` doesn't hold a `Mars` instance.
+A safer way is to use a smart cast.
+
+#### Smart casting
+
+```v oksyntax
+if w is Mars {
+    assert typeof(w).name == 'Mars'
+    if w.dust_storm() {
+        println('bad weather!')
+    }
+}
+```
+`w` has type `Mars` inside the body of the `if` statement. This is 
+known as *flow-sensitive typing*. You can also specify a variable name:
+
+```v oksyntax
+if w is Mars as mars {
+    assert typeof(w).name == 'World'
```

Comment:
Shouldn't this be `assert typeof( mars ).name == 'Mars'`?

Reply:
Ah, I see, you didn't want to demonstrate anything about `mars` but the fact, that `w` keeps its type. Ok, then it's fine :wink:.